packages/x-components/src/x-modules/url/components/__tests__/url-handler.spec.ts

```diff
@@ -116,71 +107,60 @@ describe('testing UrlHandler component', () => {
     popstateUrlWithParams('query=lego&page=2');
     popstateUrlWithParams('query=playmobil&page=3');
 
-    expect(eventSpy).toHaveBeenNthCalledWith(1, {
-      ...initialUrlState.params,
+    expect(eventSpy).toHaveBeenNthCalledWith(2, {
+      ...initialUrlState,
       query: 'lego',
       page: 2
     } as UrlParams);
 
-    expect(eventSpy).toHaveBeenNthCalledWith(2, {
-      ...initialUrlState.params,
+    expect(eventSpy).toHaveBeenNthCalledWith(3, {
+      ...initialUrlState,
       query: 'playmobil',
       page: 3
     } as UrlParams);
   });
 
   it('ignores `extra params` that are not configured in the component', () => {
-    const { on, loadUrlWithParams } = renderUrlHandler({
-      template: '<UrlHandler store="store" />'
+    const { on } = renderUrlHandler({
+      template: '<UrlHandler store="store" />',
+      urlParams: 'query=lego&page=2&warehouse=111&store=222'
     });
     const eventSpy = jest.fn();
     on('ParamsLoadedFromUrl').subscribe(eventSpy);
 
-    loadUrlWithParams('query=lego&page=2&warehouse=111&store=222');
-    loadUrlWithParams('query=playmobil&page=3&warehouse=111');
-
     expect(eventSpy).toHaveBeenNthCalledWith(1, {
-      ...initialUrlState.params,
+      ...initialUrlState,
       query: 'lego',
       page: 2,
       store: '222'
     });
-    expect(eventSpy).toHaveBeenNthCalledWith(2, {
-      ...initialUrlState.params,
-      query: 'playmobil',
-      page: 3
-    });
   });
 
   it('allows to configure the URL keys names', () => {
-    const { on, loadUrlWithParams } = renderUrlHandler({
-      template: '<UrlHandler query="q" page="p" />'
+    const { on } = renderUrlHandler({
+      template: '<UrlHandler query="q" page="p" />',
+      urlParams: 'q=lego&p=2'
     });
     const eventSpy = jest.fn();
     on('ParamsLoadedFromUrl').subscribe(eventSpy);
 
-    loadUrlWithParams('q=lego&p=2');
-    loadUrlWithParams('q=playmobil&p=3');
-
     expect(eventSpy).toHaveBeenNthCalledWith(1, {
-      ...initialUrlState.params,
+      ...initialUrlState,
       query: 'lego',
       page: 2
     });
-    expect(eventSpy).toHaveBeenNthCalledWith(2, {
-      ...initialUrlState.params,
-      query: 'playmobil',
-      page: 3
-    });
   });
 
   it('changes the URL params when `PushableUrlStateChanged` is emitted', () => {
-    const { emit, getCurrentUrlParams } = renderUrlHandler();
+    const { emit, getCurrentUrlParams } = renderUrlHandler({
+      template: '<UrlHandler store="store" />'
+    });
     emit('PushableUrlStateChanged', {
-      ...initialUrlState.params,
+      ...initialUrlState,
       query: 'lego',
       page: 2,
-      store: '111'
+      store: '111',
+      warehouse: '222'
     });
     const urlSearchParams = getCurrentUrlParams();
```

Comment:
Add a expect to check that the warehouse is null:

    expect(urlSearchParams.get('warehouse')).toBeNull();